src/swarm/neo/client/RequestOnConn.d

```diff
@@ -782,6 +782,11 @@ public class RequestOnConn: RequestOnConnBase, IRequestOnConn
         if (!this.connection)
             return false;
 
+        auto client_connection = cast(Connection)this.connection;
+        assert(client_connection);
```

Comment:
I wonder if this assertion is redundant. Wouldn't the method return when checking `if (!this.connection)`?

Reply:
I'm here checking if the cast succeeded, though.

Reply:
right, I thought it wasn't needed